src/sim_engine/filesystem.rs

```diff
@@ -25,4 +25,11 @@ impl Filesystem for SimFilesystem {
         };
         Box::new(filesystem_copy)
     }
+    fn get_id(&self) -> String {
+        self.mount_point.clone()
+    }
+
+    fn eq(&self, other: &Filesystem) -> bool {
+        self.get_id() == other.get_id()
```

Comment:
I did this because we agree that mountpoint is a bad ID.  So, when we add a good one, we just update get_id().  Not sure I see why comparing mountpoing is better here.